File: exemples/intro/factorielle/Factorielle1.java
/** 
    Factorielle, utilisation d'une boucle
**/
class Factorielle1 {
    public static void main(String[] args) {
        int n = (int)Integer.valueOf(args[0]);
        int f = factorielle(n);
        System.out.printf("Factorielle : %d%n", f);
    }
    
    public static int factorielle(int n){
        if(n == 0){
            return 1;
        }
        int res = 1;
        for(int i=1; i <= n; i++){
            res *= i; // res = res * i
        }
        return res;
    }
    
}

File: exemples/intro/factorielle/Factorielle1.py
# -*- encoding: utf-8 -*-
'''
    Factorielle, utilisation d'une boucle
'''
import sys

def factorielle1(n):
    if n == 0:
        return 1
    res = 1
    for i in range(1, n+1):
        res = res * i
    return res

def main():
    n = int(sys.argv[1])
    f = factorielle1(n)
    print f

main()
